caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -96,9 +97,30 @@ func (st ServerType) Setup(originalServerBlocks []caddyfile.ServerBlock,
 			"{tls_client_serial}", "{http.request.tls.client.serial}",
 			"{tls_client_subject}", "{http.request.tls.client.subject}",
 		)
+
+		// these are placeholders that allow a user-defined final
+		// parameters, but we still want to provide a shorthand
+		// for those, so we use a regexp to replace
+		regexpReplacements := []struct {
+			search  string
+			replace string
+		}{
+			{search: `{query\.([\w-]*)}`, replace: "{http.request.uri.query.$1}"},
+			{search: `{labels\.([\w-]*)}`, replace: "{http.request.host.labels.$1}"},
+			{search: `{header\.([\w-]*)}`, replace: "{http.request.header.$1}"},
+			{search: `{path\.([\w-]*)}`, replace: "{http.request.uri.path.$1}"},
+			{search: `{r\.([\w-]*)\.([\w-]*)}`, replace: "{http.regexp.$1.$2}"},
```

Comment:
I'm not certain if `[\w-]` (a.k.a. `[a-zA-Z0-9_-]`) is the right set of chars to allow. Maybe that's too restrictive. What else would appear there?

Reply:
That ought to cover 99.9% of use cases, I think.